src/frame/worker/crypto/recrypt/RecryptWasm.ts

```diff
@@ -14,6 +14,22 @@ const {SALT_LENGTH, PBKDF2_ITERATIONS} = CryptoConstants;
 type Plaintext = Uint8Array;
 let RecryptApi: Recrypt.Api256;
 
+/**
+ * Augments UserPrivateKey and return both the augmented private key and the augmentation factor.
+ * @param userPrivateKey current private key to be augmented
+ * @param augmentatioFactor
+ */
+export const getRotationResults = (userPrivateKey: Uint8Array): Future<Error, {newPrivateKey: Uint8Array; augmentationFactor: Uint8Array}> => {
+    return generateKeyPair().map((data) => {
+        const augmentationFactor = data.privateKey;
+        const newPrivateKey = Recrypt.subtractPrivateKeys(userPrivateKey, augmentationFactor);
+        return {
+            newPrivateKey,
+            augmentationFactor,
+        };
```

Comment:
There's really no need for these local variables, I'd just do

```
return {
  augmentationFactor: data.privateKey,
  newPrivateKey: Recrypt.subtractPrivateKeys(userPrivateKey, augmentationFactor)
}
```

Reply:
I did change this but not to your exact suggestion. The augmentationFactor is provided by generateKeyPair() as a privateKey.

Reply:
I was slightly wrong in my code sample since the `augmentationFactor` variable doesn't exist. Here's what I was trying to say this block of code can be.

```
return generateKeyPair().map(({privateKey) => ({
	augmentationFactor: privateKey,
    newPrivateKey: Recrypt.subtractPrivateKeys(userPrivateKey, privateKey)
}));
```